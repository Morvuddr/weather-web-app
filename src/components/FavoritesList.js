import React from 'react';
import * as PropTypes from 'prop-types';
import './Favorites.css';
import FavCityHeader from './FavCityHeader';
import WeatherInfo from './WeatherInfo';
import { Loader } from './Loader';

const FavoritesList = ({ cities, onRemove, isLoading }) => (
    <div className='favList'>
        {
            cities.map((city) => (
                <div key={city.id}>
                    <FavCityHeader onRemove={onRemove} city={city}/>
                    {!isLoading.includes(city.id) ? <WeatherInfo weather={city}/> : <Loader/>}
                </div>
            ))
        }
    </div>
);

FavoritesList.propTypes = {
    cities: PropTypes.arrayOf(PropTypes.object),
    isLoading: PropTypes.arrayOf(PropTypes.string),
    onRemove: PropTypes.func
};

FavoritesList.defaultProps = {
    cities: [],
    isLoading: [],
    onRemove: f => f
};


export default FavoritesList;