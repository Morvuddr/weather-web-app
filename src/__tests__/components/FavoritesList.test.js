import FavoritesList from '../../components/FavoritesList';
import React from 'react';
import renderer from 'react-test-renderer';

let cities;
beforeEach(() => {
    cities = [
        {
            id: '1',
            name: 'Moscow',
            img: 'abcd',
            temperature: 'temp',
            wind: 'wind',
            cloudiness: 'cloudiness',
            pressure: 'pressure',
            humidity: 'humidity',
            location: '[location]'
        },
        {
            id: '2',
            name: 'Kemerovo',
            img: '1234',
            temperature: 'temp',
            wind: 'wind',
            cloudiness: 'cloudiness',
            pressure: 'pressure',
            humidity: 'humidity',
            location: '[location]'
        }
    ];
});
describe('FavoritesList component', () => {

    test('matches the snapshot when cities loaded', () => {

        const tree = renderer.create(
            <FavoritesList cities={cities} isLoading={[]} onRemove={f => f} errors={[]}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


    test('matches the snapshot when cities are loading', () => {

        const isLoading = ['1'];

        const tree = renderer.create(
            <FavoritesList cities={cities} isLoading={isLoading} onRemove={f => f} errors={[]}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when on error', () => {

        const errors = ['2'];

        const tree = renderer.create(
            <FavoritesList cities={cities} isLoading={[]} onRemove={f => f} errors={errors}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});