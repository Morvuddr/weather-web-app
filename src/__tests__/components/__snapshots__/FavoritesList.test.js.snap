// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList component matches the snapshot when cities are loading 1`] = `
<div
  className="FavList"
>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="abcd"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="Loader"
    >
      <h1>
        Подождите, данные загружаются
      </h1>
      <div
        className="Spinner"
      />
    </div>
  </div>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Kemerovo
      </h1>
      <img
        alt="Иконка погоды"
        src="1234"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          wind
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          cloudiness
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          pressure
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          humidity
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          [location]
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesList component matches the snapshot when cities loaded 1`] = `
<div
  className="FavList"
>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="abcd"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          wind
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          cloudiness
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          pressure
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          humidity
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          [location]
        </h4>
      </div>
    </div>
  </div>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Kemerovo
      </h1>
      <img
        alt="Иконка погоды"
        src="1234"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          wind
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          cloudiness
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          pressure
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          humidity
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          [location]
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`FavoritesList component matches the snapshot when on error 1`] = `
<div
  className="FavList"
>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Moscow
      </h1>
      <img
        alt="Иконка погоды"
        src="abcd"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          wind
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          cloudiness
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          pressure
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          humidity
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          [location]
        </h4>
      </div>
    </div>
  </div>
  <div
    className="FavoriteCity"
  >
    <div
      className="favCityHeader"
    >
      <h1>
        Kemerovo
      </h1>
      <img
        alt="Иконка погоды"
        src="1234"
      />
      <h1>
        temp
      </h1>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="Error"
    >
      <h1>
        Не удалось загрузить данные
      </h1>
    </div>
  </div>
</div>
`;
